src/WinRT.Runtime/Interop/IMarshal.net5.cs

```diff
@@ -0,0 +1,271 @@
+            {
+                try
+                {
+                    EnsureHasFreeThreadedMarshaler();
+                    t_freeThreadedMarshaler.ReleaseMarshalData(pStm);
+                }
+                catch (Exception ex)
+                {
+                    return Marshal.GetHRForException(ex);
+                }
+                return 0;
+            }
+
+            [UnmanagedCallersOnly(CallConvs = new[] { typeof(CallConvStdcall) })]
+            private static int Do_Abi_DisconnectObject_5(IntPtr thisPtr, uint dwReserved)
+            {
+                try
+                {
+                    EnsureHasFreeThreadedMarshaler();
+                    t_freeThreadedMarshaler.DisconnectObject(dwReserved);
+                }
+                catch (Exception ex)
+                {
+                    return Marshal.GetHRForException(ex);
+                }
+                return 0;
+            }
+        }
+
+        private readonly ObjectReference<IUnknownVftbl> _obj;
+        public IObjectReference ObjRef { get => _obj; }
+        public IntPtr ThisPtr => _obj.ThisPtr;
+        public IMarshal(IObjectReference obj)
+        {
+            _obj = obj.As<IUnknownVftbl>(IID);
+        }
+
+        public unsafe void GetUnmarshalClass(Guid* riid, IntPtr pv, MSHCTX dwDestContext, IntPtr pvDestContext, MSHLFLAGS mshlFlags, Guid* pCid)
+        {
+            IntPtr thisPtr = ThisPtr;
+            Marshal.ThrowExceptionForHR(((delegate* unmanaged[Stdcall]<IntPtr, Guid*, IntPtr, MSHCTX, IntPtr, MSHLFLAGS, Guid*, int>)(*(void***)thisPtr)[3])(thisPtr, riid, pv, dwDestContext, pvDestContext, mshlFlags, pCid));
+        }
+
+        public unsafe void GetMarshalSizeMax(Guid* riid, IntPtr pv, MSHCTX dwDestContext, IntPtr pvDestContext, MSHLFLAGS mshlflags, uint* pSize)
+        {
+            IntPtr thisPtr = ThisPtr;
+            Marshal.ThrowExceptionForHR(((delegate* unmanaged[Stdcall]<IntPtr, Guid*, IntPtr, MSHCTX, IntPtr, MSHLFLAGS, uint*, int>)(*(void***)thisPtr)[4])(thisPtr, riid, pv, dwDestContext, pvDestContext, mshlflags, pSize));
+        }
+
+        public unsafe void MarshalInterface(IntPtr pStm, Guid* riid, IntPtr pv, MSHCTX dwDestContext, IntPtr pvDestContext, MSHLFLAGS mshlflags)
+        {
+            IntPtr thisPtr = ThisPtr;
+            Marshal.ThrowExceptionForHR(((delegate* unmanaged[Stdcall]<IntPtr, IntPtr, Guid*, IntPtr, MSHCTX, IntPtr, MSHLFLAGS, int>)(*(void***)thisPtr)[5])(thisPtr, pStm, riid, pv, dwDestContext, pvDestContext, mshlflags));
+        }
+
+        public unsafe void UnmarshalInterface(IntPtr pStm, Guid* riid, IntPtr* ppv)
+        {
+            IntPtr thisPtr = ThisPtr;
+            Marshal.ThrowExceptionForHR(((delegate* unmanaged[Stdcall]<IntPtr, IntPtr, Guid*, IntPtr*, int>)(*(void***)thisPtr)[6])(thisPtr, pStm, riid, ppv));
+        }
+
+        public unsafe void ReleaseMarshalData(IntPtr pStm)
+        {
+            IntPtr thisPtr = ThisPtr;
```

Comment:
same here and below.

Reply:
Fixed in https://github.com/microsoft/CsWinRT/pull/1597/commits/942670d67698e6b3a29d2fc39eea9172413d310e.